pkgs/development/libraries/onnxruntime/default.nix

```diff
@@ -152,10 +205,24 @@ stdenv.mkDerivation rec {
     "-Donnxruntime_USE_FULL_PROTOBUF=OFF"
   ] ++ lib.optionals pythonSupport [
     "-Donnxruntime_ENABLE_PYTHON=ON"
+  ] ++ lib.optionals cudaSupport [
+    "-DFETCHCONTENT_SOURCE_DIR_CUTLASS=${cutlass}"
+    "-Donnxruntime_USE_CUDA=ON"
+    "-Donnxruntime_CUDNN_HOME=${cuda-redist}"
+    "-DCMAKE_CUDA_ARCHITECTURES=${cudaArchitecturesString}"
   ];
 
   doCheck = true;
 
+  checkPhase = ''
+    true
+  '';
```

Comment:
why this combo? why not disable check phase?

Reply:
> why this combo? why not disable check phase?

Actually I think it's to disable check phase 2333

Reply:
why not `doCheck = false;`?

Reply:
Indeed it's a hack to disable the check phase but still have check dependencies available, I briefly mention it under 2. in my original comment. The update in #258392 [passes gtest explicitly](https://github.com/NixOS/nixpkgs/pull/258392/files#diff-7ac55e78d519c234b2652de54ba5e5c103a50d56153e2ae0a5b2d975dbc6ba9dR144) so I'm hopeful either that fixes it, or gtest should be moved to nativeBuldDependencies. Either way, this is a hack that will need to be fixed.

Reply:
ahhhh ok let's leave a code comment explaining that then!

Reply:
I've managed to fix it, no need to document a hack anymore :)